File: VB/MvvmSample/ViewModel/MainViewModel.vb
Imports MvvmSample.Model
Imports System.Linq

Namespace MvvmSample.ViewModel

    Friend Class MainViewModel

        Private _Chart As ChartViewModel, _DataProvider As WeatherProvider

        Public Property Chart As ChartViewModel
            Get
                Return _Chart
            End Get

            Private Set(ByVal value As ChartViewModel)
                _Chart = value
            End Set
        End Property

        Public Property DataProvider As WeatherProvider
            Get
                Return _DataProvider
            End Get

            Private Set(ByVal value As WeatherProvider)
                _DataProvider = value
            End Set
        End Property

        Public Sub New()
            DataProvider = New XmlWeatherProvider("Data/WeatherData.xml")
            Dim temperaturePane As PaneViewModel = New PaneViewModel(showXAxis:=False)
            Dim pressurePane As PaneViewModel = New PaneViewModel(showXAxis:=False)
            Dim humidityPane As PaneViewModel = New PaneViewModel(showXAxis:=True)
            Dim temperatureLegend As LegendViewModel = New LegendViewModel(dockTarget:=temperaturePane)
            Dim pressureLegend As LegendViewModel = New LegendViewModel(dockTarget:=pressurePane)
            Dim humidityLegend As LegendViewModel = New LegendViewModel(dockTarget:=humidityPane)
            Dim xAxis As XAxisViewModel = New XAxisViewModel With {.MinValue = Enumerable.First(DataProvider.WeatherInfos).Timestamp, .MaxValue = Enumerable.ElementAt(DataProvider.WeatherInfos, 10).Timestamp}
            Dim temperatureYAxis As YAxisViewModel = New YAxisViewModel(title:="Temperature, F", constantLines:=Nothing)
            Dim pressureYAxis As YAxisViewModel = New YAxisViewModel(title:="Pressure, mbar", constantLines:=Nothing)
            Dim humidityYAxis As YAxisViewModel = New YAxisViewModel(title:="Humidity, %", constantLines:=New ConstantLineViewModel() {New ConstantLineViewModel(title:=String.Empty, value:=60.0)})
            Dim temperatureSeries As SeriesViewModel = New SeriesViewModel(name:="Temperature", type:=SeriesType.Line, argumentName:="Timestamp", valueName:="Temperature", legend:=temperatureLegend, pane:=temperaturePane, yAxis:=temperatureYAxis)
            Dim pressureSeries As SeriesViewModel = New SeriesViewModel(name:="Pressure", type:=SeriesType.Area, argumentName:="Timestamp", valueName:="Pressure", legend:=pressureLegend, pane:=pressurePane, yAxis:=pressureYAxis)
            Dim humiditySeries As SeriesViewModel = New SeriesViewModel(name:="Humidity", type:=SeriesType.Bar, argumentName:="Timestamp", valueName:="RelativeHumidity", legend:=humidityLegend, pane:=humidityPane, yAxis:=humidityYAxis)
            Chart = New ChartViewModel(series:=New SeriesViewModel() {temperatureSeries, pressureSeries, humiditySeries}, panes:=New PaneViewModel() {temperaturePane, pressurePane, humidityPane}, xAxis:=xAxis, yAxes:=New YAxisViewModel() {temperatureYAxis, pressureYAxis, humidityYAxis}, legends:=New LegendViewModel() {temperatureLegend, pressureLegend, humidityLegend})
        End Sub
    End Class
End Namespace

File: CS/MvvmSample/ViewModel/MainViewModel.cs
using MvvmSample.Model;
using System;
using System.Linq;

namespace MvvmSample.ViewModel {
    class MainViewModel {
        public ChartViewModel Chart { get; private set; }

        public WeatherProvider DataProvider { get; private set; }

        public MainViewModel() {
            DataProvider = new XmlWeatherProvider("Data/WeatherData.xml");


            PaneViewModel temperaturePane = new PaneViewModel(
                    showXAxis: false
                );
            PaneViewModel pressurePane = new PaneViewModel(
                    showXAxis: false
                );
            PaneViewModel humidityPane = new PaneViewModel(
                showXAxis: true
            );

            LegendViewModel temperatureLegend = new LegendViewModel(dockTarget: temperaturePane);
            LegendViewModel pressureLegend = new LegendViewModel(dockTarget: pressurePane);
            LegendViewModel humidityLegend = new LegendViewModel(dockTarget: humidityPane);


            XAxisViewModel xAxis = new XAxisViewModel {
                MinValue = DataProvider.WeatherInfos.First().Timestamp,
                MaxValue = DataProvider.WeatherInfos.ElementAt(10).Timestamp
            };

            YAxisViewModel temperatureYAxis = new YAxisViewModel(
                    title: "Temperature, F",
                    constantLines: null
                );
            YAxisViewModel pressureYAxis = new YAxisViewModel(
                    title: "Pressure, mbar",
                    constantLines: null
                );
            YAxisViewModel humidityYAxis = new YAxisViewModel(
                    title: "Humidity, %",
                    constantLines: new ConstantLineViewModel[] {
                        new ConstantLineViewModel(
                            title: String.Empty, 
                            value: 60.0)
                    }
                );

            SeriesViewModel temperatureSeries = new SeriesViewModel(
                    name: "Temperature",
                    type: SeriesType.Line,
                    argumentName: "Timestamp",
                    valueName: "Temperature",
                    legend: temperatureLegend,
                    pane: temperaturePane,
                    yAxis: temperatureYAxis
                );
            SeriesViewModel pressureSeries = new SeriesViewModel(
                    name: "Pressure",
                    type: SeriesType.Area,
                    argumentName: "Timestamp",
                    valueName: "Pressure",
                    legend: pressureLegend,
                    pane: pressurePane,
                    yAxis: pressureYAxis
                );
            SeriesViewModel humiditySeries = new SeriesViewModel(
                name: "Humidity",
                type: SeriesType.Bar,
                argumentName: "Timestamp",
                valueName: "RelativeHumidity",
                legend: humidityLegend,
                pane: humidityPane,
                yAxis: humidityYAxis);

            Chart = new ChartViewModel(
                series: new SeriesViewModel[] { temperatureSeries, pressureSeries, humiditySeries },
                panes: new PaneViewModel[] { temperaturePane, pressurePane, humidityPane },
                xAxis: xAxis,
                yAxes: new YAxisViewModel[] { temperatureYAxis, pressureYAxis, humidityYAxis },
                legends: new LegendViewModel[] { temperatureLegend, pressureLegend, humidityLegend }
            );
        }
    }
}
